boxsdk/client/client.py

```diff
@@ -1570,46 +1631,71 @@ def create_sign_request(
         :returns:
             A dictionary representing a created SignRequest
         """
-        url = self._session.get_url('sign_requests')
+        return self.__create_sign_request(
+            signers, files, parent_folder_id, prefill_tags, are_reminders_enabled, are_text_signatures_enabled, days_valid, email_message,
+            email_subject, external_id, is_document_preparation_needed, redirect_url, declined_redirect_url, template_id)
 
-        body = {
-            'source_files': files,
-            'signers': signers,
-            'parent_folder': {
-                'id': parent_folder_id,
-                'type': 'folder'
-            }
-        }
-
-        if prefill_tags:
-            body['prefill_tags'] = prefill_tags
-        if are_reminders_enabled:
-            body['are_reminders_enabled'] = are_reminders_enabled
-        if are_text_signatures_enabled:
-            body['are_text_signatures_enabled'] = are_text_signatures_enabled
-        if days_valid:
-            body['days_valid'] = days_valid
-        if email_message:
-            body['email_message'] = email_message
-        if email_subject:
-            body['email_subject'] = email_subject
-        if external_id:
-            body['external_id'] = external_id
-        if is_document_preparation_needed:
-            body['is_document_preparation_needed'] = is_document_preparation_needed
-        if redirect_url:
-            body['redirect_url'] = redirect_url
-        if declined_redirect_url:
-            body['declined_redirect_url'] = declined_redirect_url
-        if template_id:
-            body['template_id'] = template_id
+    @api_call
+    def create_sign_request_v2(
+            self,
+            signers: Iterable,
+            files: Optional[Iterable] = None,
+            parent_folder_id: Optional[str] = None,
+            prefill_tags: Optional[Iterable] = None,
+            are_reminders_enabled: Optional[bool] = None,
+            are_text_signatures_enabled: Optional[bool] = None,
+            days_valid: Optional[str] = None,
+            email_message: Optional[Iterable] = None,
+            email_subject: Optional[str] = None,
+            external_id: Optional[str] = None,
+            is_document_preparation_needed: Optional[bool] = None,
+            redirect_url: Optional[str] = None,
+            declined_redirect_url: Optional[str] = None,
+            template_id: Optional[str] = None,
+    ) -> dict:
```

Comment:
-> 'SignRequest'
(I know it was already like that but we can fix it ;)

Reply:
apply to all 3 functions

Reply:
sure